cli/azd/cmd/package.go

```diff
@@ -144,15 +144,10 @@ func (pa *packageAction) Run(ctx context.Context) (*actions.ActionResult, error)
 			continue
 		}
 
-		packageTask := pa.serviceManager.Package(ctx, svc, nil)
-		go func() {
-			for packageProgress := range packageTask.Progress() {
-				progressMessage := fmt.Sprintf("Packaging service %s (%s)", svc.Name, packageProgress.Message)
-				pa.console.ShowSpinner(ctx, progressMessage, input.Step)
-			}
-		}()
-
-		packageResult, err := packageTask.Await()
+		pa.serviceManager.SetProgressDisplay(func(msg string) {
+			pa.console.ShowSpinner(ctx, fmt.Sprintf("Packaging service %s (%s)", svc.Name, msg), input.Step)
+		})
```

Comment:
Keeping the display logic within each command creates more of a maintenance issue for when we want to make global changes to how UX messages are displayed with regard to different operations.  Each command needs to be updated seperately.

Reply:
Can you provide a more elaborate example? The command to me, represents the running application. If not at this layer, what do you see owning what should be printed out?